Mage/src/main/java/mage/game/jumpstart/JumpstartPoolGenerator.java

```diff
@@ -42,46 +40,64 @@
    * packed in an additional plastic wrap, with a face card (the "Pack Summary card") that indicates the theme and
    * the color of the half-deck."
-   *
-   * Source:  https://mtg.gamepedia.com/Jumpstart
-   *
+     *
+     * Source: https://mtg.gamepedia.com/Jumpstart
+     *
    * Announcement: https://magic.wizards.com/en/articles/archive/news/introducing-jumpstart-new-way-play-magic-2020-02-20
    * Card Pool: https://magic.wizards.com/en/articles/archive/card-image-gallery/jumpstart
    * Deck Lists: https://magic.wizards.com/en/articles/archive/feature/jumpstart-decklists-2020-06-18
-   */
-
-  private static final String RESOURCE_NAME = "jumpstart/jumpstart.txt";
-  private static final List<JumpstartPack> JUMPSTART_PACKS;
+     */
+    private static final String RESOURCE_NAME = "jumpstart/jumpstart.txt";
+    private static List<JumpstartPack> JUMPSTART_PACKS;
 
-  static {
-    try {
-      CharSource source = Resources.asCharSource(Resources.getResource(RESOURCE_NAME), Charsets.UTF_8);
-      List<JumpstartPack> packs = new ArrayList<>();
-      JumpstartPack pack = new JumpstartPack();
-      packs.add(pack);
-      for (String line : source.readLines()) {
-        if (line.isEmpty()) {
-          if (!pack.isEmpty()) {
-            pack = new JumpstartPack();
+    private static void setupPacks(String jumpstartPacks, boolean useDefault) {
+        try {
+            CharSource source;
+            if (useDefault) {
+                source = Resources.asCharSource(Resources.getResource(RESOURCE_NAME), Charsets.UTF_8);
+            } else {
+                source = CharSource.wrap(jumpstartPacks);
+            }
+            List<JumpstartPack> packs = new ArrayList<>();
+            JumpstartPack pack = new JumpstartPack();
             packs.add(pack);
-          }
-        } else if (line.startsWith("#")) {
-          // skip comment
-        } else {
-          String[] ls = line.split(" ", 4);
-          int quantity = Integer.parseInt(ls[0]);
-          for (int i = 0; i < quantity; i++) {
-            pack.add(new DeckCardInfo(ls[3], ls[2], ls[1]));
-          }
+            for (String line : source.readLines()) {
+                if (line.isEmpty()) {
+                    if (!pack.isEmpty()) {
+                        pack = new JumpstartPack();
+                        packs.add(pack);
+                    }
+                } else if (line.startsWith("#")) {
+                    // skip comment
+                } else {
+                    String[] ls = line.split(" ", 4);
+                    int quantity = Integer.parseInt(ls[0]);
+                    for (int i = 0; i < quantity; i++) {
+                        pack.add(new DeckCardInfo(ls[3], ls[2], ls[1]));
+                    }
+                }
+            }
+            JUMPSTART_PACKS = Collections.unmodifiableList(packs);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
```

Comment:
What happen on wrong formatted file? Is tourney will be closed and user will see the error message?

Reply:
If the user uploads an incorrectly formatted jumpstart cube pack file, it will give basically incorrect packs and not do the openboosters call correctly.  I think that's ok though as it kind of indicates something went wrong.